js/ascendex.js

```diff
@@ -431,91 +431,146 @@ module.exports = class ascendex extends Exchange {
                 'amount': this.safeNumber (market, 'lotSize'),
                 'price': this.safeNumber (market, 'tickSize'),
             };
             const status = this.safeString (market, 'status');
             const active = (status === 'Normal');
-            const type = ('useLot' in market) ? 'spot' : 'future';
+            const type = ('useLot' in market) ? 'spot' : 'swap';
             const spot = (type === 'spot');
-            const future = (type === 'future');
+            const swap = (type === 'swap');
             const margin = this.safeValue (market, 'marginTradable', false);
-            let symbol = id;
-            if (!future) {
-                symbol = base + '/' + quote;
+            const contract = swap;
+            const derivative = contract;
+            const linear = contract ? true : undefined;
+            let minQty = this.safeNumber (market, 'minQty');
+            let maxQty = this.safeNumber (market, 'maxQty');
+            let minPrice = this.safeNumber (market, 'tickSize');
+            let maxPrice = undefined;
+            let symbol = base + '/' + quote;
+            if (contract) {
+                const lotSizeFilter = this.safeValue (market, 'lotSizeFilter');
+                minQty = this.safeNumber (lotSizeFilter, 'minQty');
+                maxQty = this.safeNumber (lotSizeFilter, 'maxQty');
+                const priceFilter = this.safeValue (market, 'priceFilter');
+                minPrice = this.safeNumber (priceFilter, 'minPrice');
+                maxPrice = this.safeNumber (priceFilter, 'maxPrice');
+                const str = this.safeString (market, 'underlying');
+                const substrings = str.split ('/');
+                baseId = substrings[0];
+                quoteId = substrings[1];
+                base = this.safeCurrencyCode (baseId);
+                quote = this.safeCurrencyCode (quoteId);
+                symbol = base + '/' + quote + ':' + settle;
             }
             const fee = this.safeNumber (market, 'commissionReserveRate');
             result.push ({
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
+                'settle': settle,
                 'baseId': baseId,
                 'quoteId': quoteId,
-                'info': market,
+                'settleId': settleId,
                 'type': type,
                 'spot': spot,
                 'margin': margin,
-                'future': future,
+                'swap': swap,
+                'future': false,
+                'option': false,
                 'active': active,
-                'precision': precision,
+                'derivative': derivative,
+                'contract': contract,
+                'linear': linear,
+                'inverse': contract ? !linear : undefined,
                 'taker': fee,
                 'maker': fee,
+                'contractSize': 1,
```

Comment:
I think the spot markets should have contract size undefined, and I'm not certain they should all be `1` so just double check that

Reply:
@Dan-krm You'lll need to get a TRC20/Bep20 USDT wallet address on ascendex for them to be able to do that

Reply:
I just looked at this link https://ascendex.com/en/futures/intro/btc-perp and noticed that all of the perpetual pairs are for contact size = 1 so I think I can just default it to 1 for all swaps